### ISSUE TITLE
Sort headers suggestions by popularity instead of alphabetically

### ISSUE THREAD

**Issue by sabi0** [esphen/RESTED#92]
I needed to add a `Content-Type` header to a request.
A suggestion list appears when typing a header name. But the suggestions are sorted alphabetically. And that is not very useful.
After I typed "Content" the needed header was still missing. While, for example, suggested `Content-MD5` is used way less often, I believe.

![image](https://user-images.githubusercontent.com/11509875/27382292-96a995b4-5686-11e7-9983-d182434dbd0d.png)

I imagine it could be not that easy to find out all the headers "usage frequency" (though, I guess, such stats could be found in the internet). But at least the most popular headers could be shown at the top of the suggestions list.

P.S. When only "Cont" is typed all the suggestions are about "Access-Control*".

**Comment by esphen** [comment]
I agree, but I couldn't find any statistics on this. If you find a list, please put it here, and I will be happy to do so!

**Comment by sabi0** [comment]
Actually, with response headers removed (#95) there won't be that many suggestions left:
https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#Request_fields

Any single input letter has less than 7 options. Except for X that is used for extensions.
So probably this ticket should be closed in favor of #95.

What needs to be done though is to make suggestions search start at word boundary instead of "anywhere inside".
I.e. typing "enc" should match "Accept-Encoding". But "co" should not. It should only match "Cache-Control", "Content-Length", etc.
What do you think?

**Comment by esphen** [comment]
Yeah, checking the boundary of every word is a good rule. It should limit the suggestions to something more useful. I also like #94, it is reminiscent of the find algorithms in programs like IntelliJ, sublime and fzf.

Here is the code for listing the suggestions, if anyone wants to have a go:
https://github.com/esphen/RESTED/blob/master/src/components/Request/HeaderNameAutosuggest.js#L10-L21

**Comment by esphen** [comment]
As mentioned in #94, I added a fuzzy search feature. The suggestions are now sorted based on the best match. Closing this issue in favor of that.
